docs/DeveloperGuide.adoc

```diff
@@ -81,6 +81,35 @@ Use case ends.
 +
 Use case resumes at step 2.
 
+=== Use case: Renaming existing tag
+
+*MSS*
+
+. User requests to list tags
+. AddressBooks shows a list of tags
+. User request to rename a specific tag in the list
+. AddressBook shows the selected tag to be renamed
+. User input a new name for the tag
+. AddressBook rename the tag to the new name.
++
+Use case ends.
+
+*Extensions*
+
+* 2a. The list is empty.
++
+Use case ends.
+
+* 3a. The given tag is invalid.
+** 3a1. AddressBook shows an error message.
++
+Use case resumes at step 2.
+
+* 5a. User input an empty string.
+** 5a1. AddressBook will not rename the tag.
++
+Use case ends.
```

Comment:
Possible alternative: Would it be better for AddressBook to show an error message, and continue the use case at Step 4?
Perhaps there is another way for user to tell AddressBook to not rename the tag, e.g. User inputs the same tag name, or User enters a specific keyword.